### ISSUE TITLE
./oryx-run.sh serving java.lang.NoClassDefFoundError: kafka/serializer/Decoder

### ISSUE THREAD

**Issue by pinguo-xudianyang** [OryxProject/oryx#283]
2016-04-11 00:55:22,800 INFO  ServingLayer:118 Starting Serving Layer ALSExample
Apr 11, 2016 12:55:23 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-nio-8080"]
Apr 11, 2016 12:55:23 AM org.apache.tomcat.util.net.NioSelectorPool getSharedSelector
INFO: Using a shared selector for servlet write/read
Apr 11, 2016 12:55:23 AM org.apache.catalina.core.StandardService startInternal
INFO: Starting service Tomcat
Apr 11, 2016 12:55:23 AM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/8.0.30
Apr 11, 2016 12:55:24 AM org.apache.catalina.core.StandardContext listenerStart
SEVERE: Error configuring application listener of class com.cloudera.oryx.lambda.serving.ModelManagerListener
java.lang.NoClassDefFoundError: kafka/serializer/Decoder
        at java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
        at java.lang.Class.getConstructor0(Class.java:3075)
        at java.lang.Class.newInstance(Class.java:412)
        at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:121)
        at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4734)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5255)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: kafka.serializer.Decoder
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 14 more

Apr 11, 2016 12:55:24 AM org.apache.catalina.core.StandardContext listenerStart
SEVERE: Skipped installing application listeners due to previous error(s)
Apr 11, 2016 12:55:24 AM org.apache.catalina.core.StandardContext startInternal
SEVERE: One or more listeners failed to start. Full details will be found in the appropriate container log file
Apr 11, 2016 12:55:24 AM org.apache.catalina.core.StandardContext startInternal
SEVERE: Context [Oryx] startup failed due to previous errors
Apr 11, 2016 12:55:24 AM org.apache.coyote.AbstractProtocol start

**Comment by srowen** [comment]
That means you don't have Kafka classes available in the usual place on your CDH cluster. Did you install the parcel and set up the topics per the docs? In general, you need to provide more information than this.

**Comment by pinguo-xudianyang** [comment]
Oh, I see, Thanks.

**Comment by liuxx532** [comment]
@srowen   I got same issue when I run oryx on cluster model, which  runs spark-submit in docker container and uses the standalone spark on CDH. All is well when run the spark-submit  on the CDH machine, however I copy the configuration to the docker container, this issue came. As u said the possible reason is the  *.jars. I added 2 jars in compute-classpath.sh **but still not work**. Like blow:
![image](https://cloud.githubusercontent.com/assets/12507976/22682412/7a54ddf8-ed4d-11e6-94c7-e544b056d5c7.png)
